exercises/practice/binary-search/.approaches/while-with-if-statements/content.md

```diff
@@ -0,0 +1,69 @@
+# `while` loop with `if` statements
+
+**binary_search.h**
+
+```c
+#ifndef BINARY_SEARCH_H
+#define BINARY_SEARCH_H
+
+#include <stddef.h>
+
+int* binary_search(int value, const int *arr, size_t length);
+
+#endif
+```
+
+**binary_search.c**
+
+```c
+#include "binary_search.h" 
+
+int* binary_search(int value, const int *arr, size_t length){
+    if (!arr) return NULL;
+    size_t left = 0, right = length;
+    while (left < right) {
+        size_t mid = left + ((right - left) / 2);
+        int look = arr[mid];
+        if (look == value) return (int *) &arr[mid];
+        if (look < value) left = mid + 1;
+        else right = mid;
+    }
+    return NULL;
+}
+```
+
+This approach starts by checking for a `NULL` input array.
+It then defines the [`size_t`][size_t] variables to keep track of the `left` and `right` positions being searched for in the array.
+The `left` is initialized to `0` and the `right` is initialized to the length passed in.
+
+The while loop will loop as long as the `left` position is less than the `right` position.
+
+Inside the loop, the middle value is set by `left` plus ((`right` - `left`) divided by `2`).
+- For example, if `left` is `0` and `right` is `10`, then the middle is calculated to `5`.
```

Comment:
Need blank line before list

Reply:
Whoa, sorry! Not used to reviewers doing that. Okay, change made.

Luckily, I think other changes I may have missed on previous exercises were fairly minor formatting issues.